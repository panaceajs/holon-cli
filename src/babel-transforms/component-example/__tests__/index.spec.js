const pluginTester = require('babel-plugin-tester');
const componentExample = require('../');
const pluginSyntaxObjectRestSpread = require('@babel/plugin-syntax-object-rest-spread');
const pluginJsx = require('@babel/plugin-syntax-jsx');
const dynamicImport = require('@babel/plugin-syntax-dynamic-import');

pluginTester({
  filename: __filename,
  plugin: componentExample,
  snapshot: true,
  babelOptions: {
    sourceType: 'module',
    plugins: [dynamicImport, pluginJsx, pluginSyntaxObjectRestSpread]
  },
  tests: [
    {
      title: 'it should add nothing when there is nothing to add.',
      fixture: '__fixtures__/sfc-without-props.js',
      pluginOptions: { componentName: 'ExampleComponent' }
    },
    {
      title:
        'it should change component name of component with holon-component-title class.',
      fixture: '__fixtures__/sfc-without-props-with-title.js',
      pluginOptions: { componentName: 'ExampleComponent' }
    },
    {
      title:
        'it should add props to props list of element with holon-component-props class.',
      fixture: '__fixtures__/sfc-with-props-with-list.js',
      pluginOptions: { componentName: 'ExampleComponent' }
    }
  ]
});
