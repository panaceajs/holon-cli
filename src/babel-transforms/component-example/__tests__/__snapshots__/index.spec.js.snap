// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`holon-component-example it should add nothing when there is nothing to add.: it should add nothing when there is nothing to add. 1`] = `
"
import React from 'react';

export default () => <div>123</div>;

      ↓ ↓ ↓ ↓ ↓ ↓

import React from 'react';

export default (() => <div>123</div>);
"
`;

exports[`holon-component-example it should add props to props list of element with holon-component-props class.: it should add props to props list of element with holon-component-props class. 1`] = `
"
import React from 'react';

export default ({ first = 10, functionProp = () => {  }}) => (<div><ul className=\\"holon-component-props\\"/></div>);

      ↓ ↓ ↓ ↓ ↓ ↓

import React from 'react';

export default (({ first = 10, functionProp = () => {} }) => <div><ul className=\\"holon-component-props\\"><li>first: {first}</li><li><button type=\\"button\\" onClick={() => {
        functionProp();
      }}>functionProp action</button></li></ul></div>);
"
`;

exports[`holon-component-example it should change component name of component with holon-component-title class.: it should change component name of component with holon-component-title class. 1`] = `
"
import React from 'react';

export default () => (<div><h2 className=\\"holon-component-title\\"></h2></div>);

      ↓ ↓ ↓ ↓ ↓ ↓

import React from 'react';

export default (() => <div><h2 className=\\"holon-component-title\\">ExampleComponent</h2></div>);
"
`;
